### ISSUE TITLE
DPE Inspector: Picking an entity in the Gradient Entity Id property of the PostFX Gradient Weight Modifier component causes Editor to crash

### ISSUE THREAD

**Issue by LB-ArturZieba** [o3de/o3de#16719]
**Describe the bug**
If `ed_enableDPE` is set to `true`, then picking an entity for the *Gradient Entity Id* property of the *PostFX Gradient Weight Modifier* component causes the Editor to crash.

It should be noted that this does not occur if `ed_enableDPE` is set to `false`.

This issue occurs on both Windows and Linux.

Call Stack:
```
Call Stack Trace:
25) 00007FF8506CDEA7 (KERNELBASE) : UnhandledExceptionFilter
24) 00007FF84FFCC888 (InProcessClient64) : (function-name not available)
23) 00007FF84FFCC4D1 (InProcessClient64) : (function-name not available)
22) 00007FF852DB5330 (ntdll) : memset
21) 00007FF852D9C876 (ntdll) : _C_specific_handler
20) 00007FF852DB221F (ntdll) : _chkstk
19) 00007FF852D614B4 (ntdll) : RtlRaiseException
18) 00007FF852DB0D2E (ntdll) : KiUserExceptionDispatcher
17) C:\o3de\Gems\GradientSignal\Code\Source\GradientSampler.cpp (141) : GradientSignal::GradientSampler::ValidatePotentialEntityId
16) C:\o3de\Code\Framework\AzCore\AzCore\RTTI\ReflectContext.h (723) : AZ::AttributeMemberFunction<AZ::Outcome<void,AZStd::basic_string<char,AZStd::char_traits<char>,AZStd::allocator> > (__cdecl GradientSignal::GradientSampler::*)(void *,AZ::Uuid const &)con
15) C:\o3de\Code\Framework\AzCore\AzCore\std\function\function_template.h (172) : AZStd::Internal::function_util::get_invoker<AZ::Outcome<void,AZStd::basic_string<char,AZStd::char_traits<char>,AZStd::allocator> > __cdecl(void *,AZ::Uuid const &),AZStd::Inter
14) C:\o3de\Code\Framework\AzCore\AzCore\std\function\function_template.h (685) : AZStd::function<AZ::Outcome<void,AZStd::basic_string<char,AZStd::char_traits<char>,AZStd::allocator> > __cdecl(void *,AZ::Uuid const &)>::operator()
13) C:\o3de\Code\Framework\AzCore\AzCore\RTTI\ReflectContext.h (455) : AZ::InvokeFromDomArrayInternal<AZ::Outcome<void,AZStd::basic_string<char,AZStd::char_traits<char>,AZStd::allocator> >,void *,AZ::Uuid const &,0,1>
12) C:\o3de\Code\Framework\AzCore\AzCore\RTTI\ReflectContext.h (541) : AZ::AttributeFunction<AZ::Outcome<void,AZStd::basic_string<char,AZStd::char_traits<char>,AZStd::allocator> > __cdecl(void *,AZ::Uuid const &)>::DomInvoke
11) C:\o3de\Code\Framework\AzFramework\AzFramework\DocumentPropertyEditor\DocumentSchema.h (557) : AZ::DocumentPropertyEditor::CallbackAttributeDefinition<AZ::Outcome<void,AZStd::basic_string<char,AZStd::char_traits<char>,AZStd::allocator> > __cdecl(void *,A
10) C:\o3de\Code\Framework\AzFramework\AzFramework\DocumentPropertyEditor\DocumentSchema.h (614) : AZ::DocumentPropertyEditor::CallbackAttributeDefinition<AZ::Outcome<void,AZStd::basic_string<char,AZStd::char_traits<char>,AZStd::allocator> > __cdecl(void *,A
 9) C:\o3de\Code\Framework\AzToolsFramework\AzToolsFramework\UI\PropertyEditor\PropertyEditorAPI_Internals.h (491) : AzToolsFramework::RpePropertyHandlerWrapper<AZ::EntityId>::OnValueChanged
 8) C:\o3de\Code\Framework\AzCore\AzCore\EBus\Internal\BusContainer.h (807) : AZ::Internal::EBusContainer<AzToolsFramework::IndividualPropertyHandlerEditNotifications,AzToolsFramework::IndividualPropertyHandlerEditNotifications,1,0>::Dispatcher<AZ::EBus<AzTo
 7) C:\o3de\Code\Framework\AzToolsFramework\AzToolsFramework\UI\PropertyEditor\PropertyManagerComponent.cpp (204) : AzToolsFramework::Components::PropertyManagerComponent::RequestWrite
 6) C:\o3de\Code\Framework\AzCore\AzCore\EBus\Internal\BusContainer.h (1360) : AZ::Internal::EBusContainer<AzToolsFramework::PropertyEditorGUIMessages,AzToolsFramework::PropertyEditorGUIMessages,0,1>::Dispatcher<AZ::EBus<AzToolsFramework::PropertyEditorGUIMe
 5) C:\o3de-packages\packages\qt-5.15.2-rev7-windows\qt\include\QtCore\qobjectdefs_impl.h (443) : QtPrivate::QFunctorSlotObject<`AzToolsFramework::EntityIdPropertyHandler::CreateGUI'::`2'::<lambda_1>,1,QtPrivate::List<AZ::EntityId>,void>::impl
 4) 00007FF8189D09AA (Qt5Core) : QObject::qt_static_metacall
 3) C:\o3de\Code\Framework\AzToolsFramework\AzToolsFramework\UI\PropertyEditor\PropertyEntityIdCtrl.cpp (408) : AzToolsFramework::PropertyEntityIdCtrl::SetCurrentEntityId
 2) C:\o3de\Code\Framework\AzToolsFramework\AzToolsFramework\UI\PropertyEditor\PropertyEntityIdCtrl.cpp (181) : AzToolsFramework::PropertyEntityIdCtrl::PickModeSelectEntity
 1) C:\o3de\Code\Framework\AzCore\AzCore\EBus\Internal\BusContainer.h (968) : AZ::Internal::EBusContainer<AzToolsFramework::EditorPickModeRequests,AzToolsFramework::EditorPickModeRequests,1,0>::Dispatcher<AZ::EBus<AzToolsFramework::EditorPickModeRequests,AzT
```

**Steps to reproduce**
1. Execute `ed_enable 1` command in the *Console*.
2. Create an entity, then add *PostFX Gradient Weight Modifier* and *PostFX Layer* components to it.
3. Click the entity picker in the *PostFX Gradient Weight Modifier* component's *Gradient Entity Id* property and click on any entity in the Viewport.

**Expected behavior**
Editor remains stable.

**Actual behavior**
Editor crashes.

**Video, Editor.log, error.log, and error.dmp files**

https://github.com/o3de/o3de/assets/86952082/a7e953db-b3b0-469f-9121-56e4f1afbf47

[Editor.log](https://github.com/o3de/o3de/files/12618228/Editor.log)
[error.log](https://github.com/o3de/o3de/files/12618229/error.log)
[error.zip](https://github.com/o3de/o3de/files/12618231/error.zip) - contains the `error.dmp` file.

**Found in Branch**
Stabilization/2310 (372b2b8)

**Commit ID from [o3de/o3de](https://github.com/o3de/o3de) Repository**
372b2b829a767a8b522b05acc4b27810dbf49ad7

**Desktop**
 - Device: PC
 - OS: Windows
 - Version 10
 - CPU AMD Ryzen 5 3600X 6-Core Processor
 - GPU AMD Radeon RX 5700 XT
 - Memory 16GB

**Comment by alexmontAmazon** [comment]
fixed by #16732

**Comment by LB-ArturZieba** [comment]
Issue verified as fixed on the Stabilization/2310 (9b90a24) branch.
